test/promap.js

```diff
@@ -0,0 +1,39 @@
+var assert = require('assert');
+
+var R = require('../source');
+var eq = require('./shared/eq');
+var Id = require('./shared/Id');
+
+
+describe.only('promap', function() {
+  var fromCharCode = String.fromCharCode;
+  function charCodeAt(s) { return s.charCodeAt(); }
+  function Costar(run) { return { promap: promap, run: run };
+    function promap(f, g) {
+      return Costar(R.pipe(R.map(f), run, g));
+    }
+  }
+
+  it(`dispatches to pronfuctor["fantasy-land/promap"]() if present`, function() {
+    var setJson = R.promap(JSON.parse, JSON.stringify)
+    var data = `{"promapped":false}`;
+    var actual = R.chain(R.applyTo(data), setJson(Id(R.assoc("promapped", true))));
+
+    eq(actual, `{"promapped":true}`);
+  });
+
+  it(`dispatches to pronfuctor.promap() if present`, function() {
+    var is1337Change = R.promap(R.multiply(100), R.equals(1337), Costar(R.sum));
+    var data = [10, 3, 0.3, 0.07];
+
+    eq(is1337Change.run(data), true);
+  });
+
+  it('composes two functions, f and g, before and after the final function respectively', function() {
+    var decodeChar = R.promap(charCodeAt, fromCharCode, R.add(-8));
+    var decodeString = R.promap(R.split(''), R.join(''), R.map(decodeChar));
+    var code = "ziuli";
+
+    eq(decodeString(code), "ramda");
+  });
+});
```

Comment:
I love these examples!

Reply:
Agreed! They helped me to see some of the usefulness 😄